pkgs/by-name/un/universal-android-debloater/package.nix

```
@@ -42,39 +44,44 @@ rustPlatform.buildRustPackage rec {
     pkg-config
   ];
 
-  nativeCheckInputs = [
-    clang
-  ];
+  nativeCheckInputs = [ clang ];
 
   preCheck = ''
     export HOME="$(mktemp -d)"
   '';
 
   postInstall = ''
-    wrapProgram $out/bin/uad-ng \
-      --prefix LD_LIBRARY_PATH : ${
-        lib.makeLibraryPath [
-          fontconfig
-          freetype
-          libglvnd
-          libxkbcommon
-          wayland
-          xorg.libX11
-          xorg.libXcursor
-          xorg.libXi
-          xorg.libXrandr
-        ]
-      } \
-      --suffix PATH : ${lib.makeBinPath [ android-tools ]}
+    wrapProgram $out/bin/uad-ng --prefix LD_LIBRARY_PATH : ${
+      lib.makeLibraryPath [
+        fontconfig
+        freetype
+        libglvnd
+        libxkbcommon
+        wayland
+        xorg.libX11
+        xorg.libXcursor
+        xorg.libXi
+        xorg.libXrandr
+      ]
+    } --suffix PATH : ${lib.makeBinPath [ android-tools ]}
   '';
 
+  checkFlags = [ "--skip=core::config::tests::test_load_configuration_file" ];
+
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgram = "${placeholder "out"}/bin/uad-ng";
+  versionCheckProgramArg = "--version";
```

```suggestion
  # no versionCheckHook, this is a GUI application
```
This check would always pass, since the version is gathered from the source directory name.
```
thread 'main' panicked at /build/universal-android-debloater-1.1.2-vendor/iced_winit-0.12.2/src/application.rs:119:10:
Create event loop: Os(OsError { line: 786, file: "/build/universal-android-debloater-1.1.2-vendor/winit-0.29.15/src/platform_impl/linux/mod.rs", error: Misc("neither WAYLAND_DISPLAY nor WAYLAND_SOCKET nor DISPLAY is set.") })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

---

I see; sad